src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -142,7 +149,7 @@ bool mongoConnect(const char* host, const char* db, const char* rplSet, const ch
           rplSetHosts.push_back(HostAndPort(hostTokens[ix]));
       }
 
-      connection = new DBClientReplicaSet(rplSet, rplSetHosts, 0);
+      connection = new DBClientReplicaSet(rplSet, rplSetHosts, timeout);
```

Comment:
Yes, I would say that this is the actual change. All the rest is the PR "plumbing" to bring the `timeout` to this place.